conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/NativeEnclaveHostInterface.kt

```diff
@@ -0,0 +1,121 @@
+package com.r3.conclave.enclave.internal
+
+import com.r3.conclave.common.EnclaveStartException
+import com.r3.conclave.common.internal.*
+import com.r3.conclave.mail.MailDecryptionException
+import com.r3.conclave.utilities.internal.getAllBytes
+import java.nio.ByteBuffer
+import java.util.*
+
+typealias StackFrame = CallInterfaceStackFrame<HostCallType>
+
+/**
+ * This class is the implementation of the [EnclaveHostInterface] for native enclaves.
+ * It has three jobs:
+ *  - Serve as the endpoint for calls to make to the host, see [com.r3.conclave.common.internal.CallInterface]
+ *  - Route calls from the host to the appropriate enclave side call handler, see [com.r3.conclave.common.internal.CallInterface]
+ *  - Handle the low-level details of the messaging protocol (ecalls and ocalls).
+ */
+class NativeEnclaveHostInterface : EnclaveHostInterface() {
+    /** In release mode we want to sanitise exceptions to prevent leakage of information from the enclave */
+    var sanitiseExceptions: Boolean = false
```

Comment:
Why can't this be a c'tor parameter?

Reply:
It's not know at construction time if we're in release mode or not.